x-pack/plugins/apm/public/store/reactReduxRequest/transactionDetails.tsx

```diff
@@ -14,11 +14,8 @@ import { IUrlParams } from '../urlParams';
 import { createInitialDataSelector } from './helpers';
 
 const ID = 'transactionDetails';
-const INITIAL_DATA = {};
```

Comment:
This had the unfortunate effect so when the transaction was `null` it defaulted to `{}`. This was good in a pre-typescript world where we had to code defensively but now it just causes problems. 
This was the case here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/public/components/app/TransactionDetails/view.tsx#L67

Since `!{}` returns true, the guard was never hit.